proposals/CSP.md

```diff
@@ -120,157 +103,40 @@ creates a `WebAssembly.Instance`.
-in a natural way within a CSP.
-
-Proposed Changes:
-* Response.url will be "hardened" to allow it (or it with a taint track bit)
-  to be trusted to carry information regarding the origin of a fetch
-  response.
-* WebAssembly compilation / instantiation requests that would
-  be allowed if they were script src's for non-inline JavaScript
-  will also be allowed for WebAssembly.
-   * This applies to hashes, or to origin whitelisting.
-* This sub-proposal only affects WebAssembly.compileStreaming
-  and WebAssembly.instatiateStreaming
-  
-### CSP Policy Application Summary
-
-The checks performed for web assembly operations is summarized as follows:
-
-Operation | default | no unsafe-eval | with wasm-unsafe-eval | with unsafe-eval and wasm-unsafe-eval 
+### Out of Scope Threats
+
+* **Bugs in the browser**. We assume correct implementations of image decoders,
+  script compilers, etc. CSP does not protect against malicious inputs that can,
+  for example, trigger buffer overflows.
+* **Resource exhaustion**. Computation performed by scripts uses memory and CPU
+  time and can therefore cause a denial of service on the browser. Protecting
+  against this is one reason site owners use CSP, but denial of service is not a
+  first order consideration for CSP. Scripts are dangerous not because of their
+  resource consumption but because of other effects that can cause.
+
+## WebAssembly and CSP
+
+As noted earlier, CSP allows publishers to control what resources are loaded into a browser from a Web application. This includes resources that are dynamically created as well as those loaded directly. There is no direct equivalent of a `script` element tag for WebAssembly modules; although it is possible that such an feature may be specified in the future. Instead, WebAssembly modules are compiled and instantiated via a JavaScript API. Thus our focus on CSP for WebAssembly is on that API.
+
+### The `HostEnsureCanCompileWasmBytes` Policy Point
+
+It is proposed that the above APIs are _gated_ by a policy point: `HostEnsureCanCompileWasmBytes`. (This is sometimes referred to as an _abstract operation_ that must be performed by the host to permit the compilation.) 
+
+#### `HostEnsureCanCompileWasmBytes`
+
+If `HostEnsureCanCompileWasmBytes` is not enabled, then the `WebAssembly.compile` and `WebAssembly.compileStreaming` functions fail with a `WebAssembly.EvalError??` exception. Otherwise, these API functions return results depending on the internal integrity of the WebAssembly module being compiled.
+
+### The `wasm-eval` source directive
+
+We recommend that a new CSP policy directive `wasm-eval` be created. If set in the headers of a page, then the  `HostEnsureCanCompileWasmBytes` policy point is enabled; which, in turn, allows the page to load, compile and instantiate WebAssembly code. The details of these abstract operations will be incorporated into a future version of the CSP specification.
```

Comment:
Why was this changed back to `wasm-eval` rather than `unsafe-wasm-eval`/`wasm-unsafe-eval`? We already settled on this in #15.

Reply:
will change to wasm-unsafe-eval